crengine/src/lvrend.cpp

```diff
@@ -2511,7 +2515,24 @@ void renderFinalBlock( ldomNode * enode, LFormattedText * txform, RenderRectAcce
         // In legacy rendering mode, we should get the same text formatting flags
         // as in CoolReader 3.2.38 and earlier, i.e. set is_block to true for
         // any block elements.
-        bool is_block = BLOCK_RENDERING_G(ENHANCED) ? rm == erm_final : style->display >= css_d_block;
+        bool is_block = rm == erm_final;
+        if (!BLOCK_RENDERING_G(ENHANCED) && !is_block) {
```

Comment:
May be now that you made it a `if`, move your 3 comment lines above ("// In legacy rendering mode...") inside it - so one can more easily skip reading comments regarding legacy hacks.